File: antlr.hello/src/main/java/cypher/grammar/my/MyCypherListener.java
package cypher.grammar.my;

import cypher.grammar.CypherBaseListener;
import cypher.grammar.CypherParser;
import cypher.grammar.CypherParser.MatchContext;
import cypher.grammar.CypherParser.ReturnContext;
import cypher.grammar.CypherParser.VariableContext;

public class MyCypherListener extends CypherBaseListener {

	@Override
	public void enterMatch(MatchContext ctx) {
		System.out.println("Enter: " + ctx.getText());
	}

	@Override
	public void exitMatch(MatchContext ctx) {
		System.out.println("Exit: " + ctx.getText());
	}

	@Override
	public void enterVariable(VariableContext ctx) {
		System.out.println("var: " + ctx.getText());
	}

	@Override
	public void enterReturn(ReturnContext ctx) {
		System.out.println("Enter: " + ctx.getText());
	}

	@Override
	public void exitReturn(ReturnContext ctx) {
		System.out.println("Enter: " + ctx.getText());
	}
}

File: kotlin/ingraph-grammar/src/main/kotlin/pkg/MyCypherListener.kt
package pkg

import cypher.grammar.CypherBaseListener
import cypher.grammar.CypherParser

class MyCypherListener : CypherBaseListener() {

    override fun enterMatch(ctx: CypherParser.MatchContext) {
        println("Enter: " + ctx.text)
    }

    override fun exitMatch(ctx: CypherParser.MatchContext) {
        println("Exit: " + ctx.text)
    }

    override fun enterVariable(ctx: CypherParser.VariableContext) {
        println("var: " + ctx.text)
    }

    override fun enterReturn(ctx: CypherParser.ReturnContext) {
        println("Enter: " + ctx.text)
    }

    override fun exitReturn(ctx: CypherParser.ReturnContext) {
        println("Exit " + ctx.text)
    }

    override fun enterWhere(ctx: CypherParser.WhereContext) {
        println("Enter: " + ctx.text)
    }

    override fun exitWhere(ctx: CypherParser.WhereContext) {
        println("Exit " + ctx.text)
    }
}
